### DISCUSSION TITLE
How to bypass the microsoft login in webdriverIO with cucumber framework

### DISCUSSION THREAD

**Discussion by Digitalteamautomation** [webdriverio/webdriverio#10373] [Q&A]
Every time i run a testcases i need to login to Microsoft even though chrome has the cookies. 
<img width="960" alt="image" src="https://github.com/webdriverio/webdriverio/assets/131886499/028545e6-64ab-48f9-a7ab-f5f925363e9f">

**Comment by portyu9** [comment]
Hi @Digitalteamautomation 

The important point is that the Chrome instance started by WebdriverIO is normally a **separate automation profile**. Cookies in your regular interactive Chrome profile are therefore not automatically available to the WDIO session.

I would avoid automating the Microsoft login UI before every scenario if this is your own application/test tenant. Prefer one of these approaches:

1. Authenticate through an application-supported test/API flow and seed the resulting application session/cookies.
2. Perform the interactive login once per WDIO worker/session rather than once per Cucumber scenario.
3. If absolutely necessary, use a dedicated persistent Chrome `user-data-dir` created only for the tests.

Don't point parallel tests at the same real Chrome profile: Chrome locks user-data directories, and persisted Microsoft tokens can expire or be invalidated by Conditional Access/MFA policies anyway.

So the existence of cookies in your normal Chrome window doesn't mean the WebdriverIO browser has those cookies.

The screenshot attached to the original discussion could not be retrieved from the current indexed page, so I did not make assumptions about the exact Microsoft screen shown. There were no replies.

Please mark this answer as accepted if it helped, thank you.
